File: WebAPI/Models/AdvertModelList.cs
using System;
using System.Collections.Generic;

namespace WebAPI.Models
{
    public class AdvertModelList
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Price { get; set; }
        public double AverageRating { get; set; }
        public DateTime EditDate { get; set; }
        public List<ImageModel> Images { get; set; }
    }
}

File: WebAPI/ClientApp/src/app/models/AdvertModelList.ts
import { ImageModel } from "./ImageModel";

export class AdvertModelList {
    public id: number;
    public name: string;
    public price: number;
    public averageRating: number;
    public editDate: Date;
    public images: ImageModel[];

    public constructor(_id: number, _name: string, _price: number, _averageRating: number, _editDate: Date, _images: ImageModel[]) {
        this.id = _id;
        this.name = _name;
        this.price = _price;
        this.averageRating = _averageRating;
        this.editDate = _editDate;
        this.images = _images;
    }
}